### PR TITLE
Refuse uploading too big .zip files

### LINKED ISSUE #840: Storing large amount of data fails
I've analyzed the `xerces` project by the `--enable-all` flag. The size of the report folder is 7.6G. When I tried to store it to the database I got an error:

```bash
store.py:333 main() - Storage failed: Central directory offset would require ZIP64 extensions
```
Problem: the maximum size of the archive file is 4GB (https://en.wikipedia.org/wiki/Zip_(file_format)#Limits).

Recommended solution:
introduce a config file:
server_config.json

maximum_storage_size: in MB
maximum_storage_reports: in number of reports

- [x] limit the maximum uploadable zip file size to 1GB? (measure run with )
- [ ] increase the report session timer to 30mins and try to starti it behind plist parsing
- [x] when storing into the zip file on the client, do not copy to tmp first, write directly to zip and set the uploadable limit to 1GB
- [x] do not parse the plist on the server side 2 times and do not parse inside open db transaction.

### PR DESCRIPTION
**Pull request by bruntib**
Fixes #840

### CODE REVIEW

File: libcodechecker/libhandlers/store.py [hunk 1]
```diff
@@ -315,6 +316,11 @@ def main(args):
 
     try:
         assemble_zip(args.input, zip_file, client)
+
+        if os.stat(zip_file).st_size > MAX_UPLOAD_SIZE:
+            LOG.error("The result list to upload is too big.")
```
Comment (csordasmarton) [inline libcodechecker/libhandlers/store.py]: Write out the maximum uploadable file size too with an exclamation mark at the end of the sentence.

File: libcodechecker/libhandlers/store.py [hunk 2]
```diff
@@ -315,6 +316,11 @@ def main(args):
 
     try:
         assemble_zip(args.input, zip_file, client)
+
+        if os.stat(zip_file).st_size > MAX_UPLOAD_SIZE:
+            LOG.error("The result list to upload is too big (max: 1GB).")
```
Comment (Xazax-hun) [inline libcodechecker/libhandlers/store.py]: Shouldn't the error message depend on the MAX_UPLOAD_SIZE variable?

**Comment by bruntib** [comment]
The others items in #840 are already satisfied, So that can be closed. To be exact the 30 sec timeout is not done yet but that shouldn't be relevant, since there is no such a long action during storage. But the timeout can be increased later if needed.

**Comment by csordasmarton** [comment]
LGTM!

**Comment by gyorb** [comment]
I've created a new issue for the missing session improvement. #1072

**Pull request merged**